packages/accounts-base/accounts_server.js

```diff
@@ -1401,6 +1401,28 @@ export class AccountsServer extends AccountsCommon {
       // XXX Maybe we should re-use the selector above and notice if the update
       //     touches nothing?
       setAttrs = { ...setAttrs, ...opts };
+
+      // Sync email from OAuth service data to user.emails array if present
+      if (serviceData.email) {
+        const emailAddress = serviceData.email;
+        const emailVerified = serviceData.verified_email || serviceData.verified || false;
+
+        // Check if the email already exists in user.emails
+        const existingUser = await this.users.findOneAsync(user._id, {
```

Comment:
I would prefer a way where we don't have to have a DB call. Can we get this information from an existing source or I think the update on line 1426 could be adjusted to have a proper update that would add to the array if it exists or create it.